# Generated by Django 3.2.7 on 2024-01-03 16:15

from django.conf import settings
from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
        ('modal_2fa', '0001_initial'),
    ]

    operations = [
        migrations.CreateModel(
            name='WebauthnCredential',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('credential_public_key', models.CharField(blank=True, max_length=9000, null=True)),
                ('credential_id', models.CharField(blank=True, max_length=9000, null=True)),
                ('created_on', models.DateTimeField(auto_now_add=True)),
                ('last_used_on', models.DateTimeField(null=True)),
                ('sign_count', models.IntegerField()),
                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='webauthn', to=settings.AUTH_USER_MODEL)),
            ],
        ),
        migrations.CreateModel(
            name='FailedLoginAttempt',
            fields=[
                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('ip_address', models.CharField(blank=True, max_length=20, null=True)),
                ('failed_attempts', models.IntegerField()),
                ('locked_time', models.DateTimeField(blank=True, null=True)),
                ('user', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='failed_login', to=settings.AUTH_USER_MODEL)),
            ],
        ),
    ]
